### PR TITLE
#35 - dry run and scrub images from description

### PR DESCRIPTION
**Pull request by cbeach47**
Supports a 'dry run' ability and scrubs images embedded in various html tags in the description.

image

Also upgraded Bb libraries to 3800.15.0 , which required upgrading Gradle and nebula.lint.

### CODE REVIEW

**Comment by cbeach47** [comment]
Due to the nature of this side project, I plan to move fairly quick on this PR and merge this evening unless folks have any concerns with the changes.

File: oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/Fixer.java
Comment (ChristianMurphy) [inline oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/Fixer.java]:
```suggestion
	if (execute && dryrun) {
```

File: oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/Fixer.java
Comment (ChristianMurphy) [inline oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/Fixer.java]:
```suggestion
	if (dryrun) {
```

File: oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/Fixer.java
Comment (ChristianMurphy) [inline oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/Fixer.java]:
```suggestion
	  utils.log(level + 1, "Migrated link persisted to database");
```

File: oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java
Comment (ChristianMurphy) [inline oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java]: indentation seems off here, the `if` appears to be inside `for` but is not indented as such

File: oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java
Comment (ChristianMurphy) [inline oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java]: Regex parsing HTML feels pretty risky.
There are a lot of quirks and edge cases in the language, would it be feasible to do these transforms using something like jsoup? https://jsoup.org/

**Review by ChristianMurphy — COMMENTED** [review]
Travis CI is flagging 100 build errors, all related to `javax.jws.*`.
Would is be possible to fix these before merging?
https://travis-ci.com/github/openequella/openEQUELLA-blackboard-integration/builds/176595380#L246-L661

**Comment by cbeach47** [comment]
The SOAP WS are being removed in Blackboard, and this release is focused on just the migration B2.  I'll update the build targets to only release that project for now.  Ideally, the standard B2 and WS won't have to be revised again, and adopters will switch over to the standard LTI flows.

File: oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java
Comment (cbeach47) [reply oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java]: I can appreciate that.  Using reluctant qualifiers and whitespace wildcards lowers the risk, and we are focused on some very specific use cases for the migration.  I'll see if we can integrate `jsoup` without too much hassle.

Comment (cbeach47) [reply oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java]: JSoup is now integrated.

File: oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java
Comment (cbeach47) [reply oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java]: Opened #37 to resolve formatting issues repo-wide.

**Comment by cbeach47** [comment]
Cleaned up the regex's and string replacements for a cleaner, more robust solution (Thanks @ChristianMurphy ).

File: oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java
Comment (ChristianMurphy) [inline oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java]: I wonder if using a prefix selector (https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors)

```
a[href^="/webapps/dych-tle-"]
```

could help simplify this step

File: oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java
Comment (ChristianMurphy) [inline oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java:441]: lower down it looks like it wants to focus just in images inside tables
```
<tr><td><img .../></td></tr>
```

would it make sense to do that selection here?

```css
tr > td > img
```
https://developer.mozilla.org/en-US/docs/Web/CSS/Child_combinator

**Review by ChristianMurphy — APPROVED** [review]
Looks good, thanks @cbeach47!
A couple small simplifications may be possible, but the approach as a whole looks solid :+1:

File: oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java
Comment (cbeach47) [reply oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java:441]: that was a nicety, but if it can't determine that the `img` is only inside a `td` and `tr`, it'll still remove the `img`

File: oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java
Comment (cbeach47) [reply oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java]: huh.  I thought I tried that.  Apparently not though.  :upside_down_face:   Switched to using the prefix selector.

File: oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java
Comment (ChristianMurphy) [reply oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java:441]: ahh, good to know

**Pull request merged**
